src/c_api.rs

```diff
@@ -420,6 +464,36 @@ pub unsafe extern "C" fn c2pa_reader_from_stream(
     }
 }
 
+/// Creates and verifies a C2paReader from manifest data and an asset stream.
+///
+/// Parameters
+/// * manifest_data: pointer to the manifest data bytes.
+/// * manifest_data_size: size of the manifest data in bytes.
+/// * format: pointer to a C string with the mime type or extension.
+/// * stream: pointer to a C2paStream.
+///
+/// # Errors
+/// Returns NULL if there were errors, otherwise returns a pointer to a ManifestStore.
+/// The error string can be retrieved by calling c2pa_error.
+///
+/// # Safety
+/// Reads from NULL-terminated C strings.
+/// The returned value MUST be released by calling c2pa_reader_free
+/// and it is no longer valid after that call.
+#[no_mangle]
+pub unsafe extern "C" fn c2pa_reader_from_manifest_data_and_stream(
+    format: *const c_char,
+    stream: *mut C2paStream,
+    manifest_data: *const c_uchar,
+    manifest_size: usize,
+) -> *mut C2paReader {
+    let format = from_cstr_or_return_null!(format);
```

Comment:
you just made me realize I need to add more null checks, but this is no worse than what I've already released.
We should add:
check_or_return_null!(stream)
check_or_return_null!(manifest_data);

Reply:
Yes, we're lacking on null checks - but I figured this could happen in the "final" C API.